### PR TITLE
Update for MP6

### PR DESCRIPTION
**Pull request by majidmostafavi**
Add Telemetry 
update for MP6

### CODE REVIEW

**Comment by majidmostafavi** [comment]
@Emily-Jiang 
I find the error  @Health is deprecated.  so I should replace it by @Liveness ,@Readiness.

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang I find the error @health is deprecated. so I should replace it by @Liveness ,@readiness.

Depending on which version of MP health you are talking about. It was only deprecated later. It was fine in the earlier version of MP Health. I suggest you to fix the test error first not to change other things.

**Comment by Emily-Jiang** [comment]
@luiseufrasio did you figure out what was the test failures and propose a update on the PR to unblock this?

**Comment by luiseufrasio** [comment]
> clipsefdn/eca

@Emily-Jiang the tests seems to be NOT running: https://github.com/eclipse/microprofile-starter/pull/494/checks
I says "Workflow runs completed with no jobs"

**Comment by Emily-Jiang** [comment]
The conflict needs to be solved before the tests can be executed. @majidmostafavi can you please resolve the conflicts and then the CI jobs can be launched?

File: src/main/java/org/eclipse/microprofile/starter/rest/APIEndpointV7.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.eclipse.microprofile.starter.rest;
+
+import javax.ws.rs.Path;
```
Comment (Emily-Jiang) [inline src/main/java/org/eclipse/microprofile/starter/rest/APIEndpointV7.java:3]:
```suggestion
import jakarta.ws.rs.Path;
```

File: src/main/java/org/eclipse/microprofile/starter/rest/APIEndpointV7.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.eclipse.microprofile.starter.rest;
+
+import jakarta.ws.rs.Path;
```
Comment (Emily-Jiang) [inline src/main/java/org/eclipse/microprofile/starter/rest/APIEndpointV7.java:3]:
```suggestion
import javax.ws.rs.Path;
```

File: src/test/resources/json_examples/v7/supportMatrix.json.segments [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{"configs":{"
+MP33":{"supportedServers":["
+]},"MP32":{"supportedServers":[
```
Comment (Emily-Jiang) [inline src/test/resources/json_examples/v7/supportMatrix.json.segments:3]: I am not sure this file. Do you need to just mention MP60?

Comment (Emily-Jiang) [reply src/test/resources/json_examples/v7/supportMatrix.json.segments:3]: @rdebusscher we are stuck on this failure. Can you please shed some lights on this? Your help is very much appreciated.

Comment (rdebusscher) [reply src/test/resources/json_examples/v7/supportMatrix.json.segments:3]: @Emily-Jiang What is the issue exactly? From the failed test run, it just seems a problem with some assertions on text values.
Not much time to go in detail for the moment.

Comment (luiseufrasio) [reply src/test/resources/json_examples/v7/supportMatrix.json.segments:3]: I found the error in the surefire-reports and changed the code to print the URL call which fails: Download failed. URI: http://127.0.0.1:9090/api/project?supportedServer=LIBERTY&selectedSpecs=CONFIG&selectedSpecs=FAULT_TOLERANCE&selectedSpecs=HEALTH_CHECKS&selectedSpecs=METRICS&selectedSpecs=OPEN_TRACING&selectedSpecs=OPEN_API&artifactId=liberty&buildTool=MAVEN expected:<200> but was:<400>

Comment (luiseufrasio) [reply src/test/resources/json_examples/v7/supportMatrix.json.segments:3]: I tried the URL locally running on liberty:
image
So after I called: `http://127.0.0.1:9080/mp-starter/api/project?supportedServer=LIBERTY&selectedSpecs=CONFIG&selectedSpecs=FAULT_TOLERANCE&selectedSpecs=HEALTH_CHECKS&selectedSpecs=METRICS&selectedSpecs=OPEN_TRACING&selectedSpecs=OPEN_API&artifactId=liberty&buildTool=MAVEN`, I received back:```
 {
	"error": "One or more selectedSpecs is not available for the given mpVersion",
	"code": "ERROR003"
}
```

Comment (Emily-Jiang) [reply src/test/resources/json_examples/v7/supportMatrix.json.segments:3]: I am not sure the version combination. 
What about the error for APITest?

**Comment by rdebusscher** [comment]
Why is there a v7 of the Api created when it is identical to v6?

**Comment by Emily-Jiang** [comment]
> Why is there a v7 of the Api created when it is identical to v6?

v7 has Telemetry added though.

File: src/test/resources/json_examples/v7/supportMatrix.json.segments (hunk 3 above)
Comment (Emily-Jiang) [reply src/test/resources/json_examples/v7/supportMatrix.json.segments:3]: Which file needs to be updated? @rdebusscher

Comment (rdebusscher) [reply src/test/resources/json_examples/v7/supportMatrix.json.segments:3]: Based on the log of the test

https://github.com/eclipse/microprofile-starter/pull/494/checks#step:7:17864

This is the failure
```
supportMatrix(org.eclipse.microprofile.starter.APITest)  Time elapsed: 0.319 sec  <<< FAILURE!
java.lang.AssertionError: Response of /7/supportMatrix 
```

And from the test method `APITest#supportMatrix`
````
    public void supportMatrix() throws FileNotFoundException {
        test(v7Matrix, "/7/supportMatrix");
````

we see that the contents of this file os compared to the response

````
v7Matrix = new File(getClass().getClassLoader().getResource("json_examples/v7/supportMatrix.json.segments").getFile());        
````

**Comment by rdebusscher** [comment]
The version of the API has nothing to do with the config op the MP versions and the options.  It is about the available endpoints, their parameters and responses.

So `/api/mpVersion/MP60`, `/api/7/mpVersion/MP60`, and , `/api/6/mpVersion/MP60` all return

````
{"specs":["CONFIG","FAULT_TOLERANCE","JWT_AUTH","METRICS","HEALTH_CHECKS","OPEN_API","REST_CLIENT","TELEMETRY","GRAPHQL"],"supportedServers":["LIBERTY"]}
````

Even the API versions 2 and higher, all return this result which includes Telemetry.

So there is no need at all to have a new API version.

**Comment by Emily-Jiang** [comment]
> The version of the API has nothing to do with the config op the MP versions and the options. It is about the available endpoints, their parameters and responses.
> 
> So `/api/mpVersion/MP60`, `/api/7/mpVersion/MP60`, and , `/api/6/mpVersion/MP60` all return
> 
> ```
> {"specs":["CONFIG","FAULT_TOLERANCE","JWT_AUTH","METRICS","HEALTH_CHECKS","OPEN_API","REST_CLIENT","TELEMETRY","GRAPHQL"],"supportedServers":["LIBERTY"]}
> ```
> 
> Even the API versions 2 and higher, all return this result which includes Telemetry.
> 
> So there is no need at all to have a new API version.

I thought the earlier version of MP (1.x-5.x) should NOT return Telemetry @rdebusscher

**Comment by rdebusscher** [comment]
Earlier versions of MP will not return Telemetry. As I already said, MP Version and API version of the Starter app are 2 independent things.

**Comment by Emily-Jiang** [comment]
> Earlier versions of MP will not return Telemetry. As I already said, MP Version and API version of the Starter app are 2 independent things.

Should the earlier API version such as 2.x return Telemetry at all?

**Comment by rdebusscher** [comment]
> > Earlier versions of MP will not return Telemetry. As I already said, MP Version and API version of the Starter app are 2 independent things.
> 
> Should the earlier API version such as 2.x return Telemetry at all?

In my opinion, yes. Those that make an integration with the API should not use latest version since that can have breaking changes at any time. Only a certain API version is stable.
And having all options in all API versions doesn't force external parties to update their integration with the MP Starter API.

**Comment by Emily-Jiang** [comment]
> > > Earlier versions of MP will not return Telemetry. As I already said, MP Version and API version of the Starter app are 2 independent things.
> > 
> > 
> > Should the earlier API version such as 2.x return Telemetry at all?
> 
> In my opinion, yes. Those that make an integration with the API should not use latest version since that can have breaking changes at any time. Only a certain API version is stable. And having all options in all API versions doesn't force external parties to update their integration with the MP Starter API.

Thanks @rdebusscher ! @majidmostafavi you don't need to create a new version of the API.

File: src/main/java/org/eclipse/microprofile/starter/rest/APIEndpointV7.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package org.eclipse.microprofile.starter.rest;
+
+import javax.ws.rs.Path;
+
+@Path("/7")
```
Comment (Emily-Jiang) [inline src/main/java/org/eclipse/microprofile/starter/rest/APIEndpointV7.java:5]: Based on my understanding from @rdebusscher , there is no need to do this change.

**Comment by breakponchito** [comment]
@Emily-Jiang  I found the way to fix the issues from the  API MicroProfile Starter but I cant add commits on this PR, it seems someone needs to give me some permission:

`
remote: Permission to majidmostafavi/microprofile-starter.git denied to breakponchito.
fatal: unable to access 'https://github.com/majidmostafavi/microprofile-starter.git/': The requested URL returned error: 403
00:12:20.257: [microprofile-starter] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false push --progress --porcelain majid refs/heads/majidMain:main
remote: Permission to majidmostafavi/microprofile-starter.git denied to breakponchito.
fatal: unable to access 'https://github.com/majidmostafavi/microprofile-starter.git/': The requested URL returned error: 403
`

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang @cesarhernandezgt I found the way to fix the issues from the API MicroProfile Starter but I can't add commits on this PR, it seems someone needs to give me some permission:
> 
> `remote: Permission to majidmostafavi/microprofile-starter.git denied to breakponchito. fatal: unable to access 'https://github.com/majidmostafavi/microprofile-starter.git/': The requested URL returned error: 403 00:12:20.257: [microprofile-starter] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false push --progress --porcelain majid refs/heads/majidMain:main remote: Permission to majidmostafavi/microprofile-starter.git denied to breakponchito. fatal: unable to access 'https://github.com/majidmostafavi/microprofile-starter.git/': The requested URL returned error: 403` probably I should need to recreate the PR

@breakponchito it was the branch from @majidmostafavi . I think you can checkout his branch and then do a new PR.

**Comment by breakponchito** [comment]
> > @Emily-Jiang @cesarhernandezgt I found the way to fix the issues from the API MicroProfile Starter but I can't add commits on this PR, it seems someone needs to give me some permission:
> > `remote: Permission to majidmostafavi/microprofile-starter.git denied to breakponchito. fatal: unable to access 'https://github.com/majidmostafavi/microprofile-starter.git/': The requested URL returned error: 403 00:12:20.257: [microprofile-starter] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false push --progress --porcelain majid refs/heads/majidMain:main remote: Permission to majidmostafavi/microprofile-starter.git denied to breakponchito. fatal: unable to access 'https://github.com/majidmostafavi/microprofile-starter.git/': The requested URL returned error: 403` probably I should need to recreate the PR
> 
> @breakponchito it was the branch from @majidmostafavi . I think you can checkout his branch and then do a new PR.
> 
> ```
> git checkout -b majidmostafavi-main main
> git pull https://github.com/majidmostafavi/microprofile-starter.git main
> ```

yes I already create a new PR here: https://github.com/eclipse/microprofile-starter/pull/499 
it seems that I need an approval from a maintainer

@Emily-Jiang Could you help me with this? 
image

**Pull request merged**